actors/miner/tests/daily_fees_test.rs

```diff
@@ -76,6 +83,218 @@ fn fee_paid_at_deadline() {
+        }
+        dlinfo
+    };
+
+    let verify_balance_change = |expected_deduction: &TokenAmount, operation: &dyn Fn()| {
+        let miner_balance_before = rt.get_balance();
+        operation();
+        let miner_balance_after = rt.get_balance();
+        assert_eq!(miner_balance_before - expected_deduction, miner_balance_after);
+    };
+
+    let submit_window_post =
+        |dlinfo: &DeadlineInfo, sectors: &Vec<SectorOnChainInfo>, post_config: PoStConfig| {
+            let partition = PoStPartition { index: pidx, skipped: make_empty_bitfield() };
+            h.submit_window_post(&rt, dlinfo, vec![partition], sectors.clone(), post_config)
+        };
+
+    // Step 1. Normal PoST but capped by reward, not daily_fee
+
+    let dlinfo = advance_to_post_deadline();
+    // configure post for power delta in the capped-upfront case
+    let cfg = if capped_upfront {
+        PoStConfig::with_expected_power_delta(&sectors_power.clone())
+    } else {
+        PoStConfig::empty() // no power delta, we've had first-post already
+    };
+    submit_window_post(&dlinfo, &sectors, cfg);
+
+    let state = h.get_state(&rt);
+    let unvested = unvested_vesting_funds(&rt, &state);
+    let available = rt.get_balance() + unvested.clone() - &state.initial_pledge;
+    let burnt_funds = day_reward.div_floor(2); // i.e. not daily_fee
+    assert!(available >= burnt_funds);
+
+    verify_balance_change(&burnt_funds, &|| {
+        h.advance_deadline(
+            &rt,
+            CronConfig {
+                burnt_funds: burnt_funds.clone(),
+                pledge_delta: burnt_funds.clone().neg(),
+                ..Default::default()
+            },
+        );
+    });
+
+    // Step 2. Advance to next deadline, fail to submit post, make sure we have faulty power, and
+    // then assert that the cap is unchanged. i.e. it includes faulty power in its calculation.
+
+    advance_to_post_deadline();
+    verify_balance_change(&burnt_funds, &|| {
+        h.advance_deadline(
+            &rt,
+            CronConfig {
+                burnt_funds: burnt_funds.clone(),
+                pledge_delta: burnt_funds.clone().neg(),
+                power_delta: Some(sectors_power.clone().neg()),
+                ..Default::default()
+            },
+        );
+    });
+
+    // Step 3. Advance to next deadline and submit post, recovering power, and pay the same capped
+    // fee.
+
+    let bf = bitfield_from_slice(&sectors.iter().map(|s| s.sector_number).collect::<Vec<u64>>());
+    h.declare_recoveries(&rt, dlidx, pidx, bf, TokenAmount::zero()).unwrap();
+    let dlinfo = advance_to_post_deadline();
+    submit_window_post(&dlinfo, &sectors, PoStConfig::with_expected_power_delta(&sectors_power));
+    verify_balance_change(&burnt_funds, &|| {
+        h.advance_deadline(
+            &rt,
+            CronConfig {
+                burnt_funds: burnt_funds.clone(),
+                pledge_delta: burnt_funds.clone().neg(),
```

Comment:
```suggestion
                pledge_delta: -burnt_funds.clone(),
```

Reply:
(I know we have a tendency to call `.neg()` all over the place, but using the actual negation operator is nicer, IMO)